src/ModelContextProtocol/Protocol/Transport/StdioClientStreamTransport.cs

```diff
@@ -0,0 +1,324 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using ModelContextProtocol.Configuration;
+using ModelContextProtocol.Logging;
+using ModelContextProtocol.Protocol.Messages;
+using ModelContextProtocol.Utils;
+using ModelContextProtocol.Utils.Json;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+
+namespace ModelContextProtocol.Protocol.Transport;
+
+/// <summary>
+/// Implements the MCP transport protocol over standard input/output streams.
+/// </summary>
+internal sealed class StdioClientStreamTransport : TransportBase
```

Comment:
What is the difference between StdioClientTransport and StdioClientStreamTransport? It's not clear to me why we have both.